ledger/ledgercore/statedelta.go

```diff
@@ -103,23 +108,40 @@ type StateDelta struct {
 }
 
 // AccountDeltas stores ordered accounts and allows fast lookup by address
-type AccountDeltas struct {
+// type AccountDeltas struct {
+// 	// Actual data. If an account is deleted, `accts` contains a balance record
+// 	// with empty `AccountData`.
+// 	accts []basics.BalanceRecord
+// 	// cache for addr to deltas index resolution
+// 	acctsCache map[basics.Address]int
+// }
+
+type newBalanceRecord struct {
+	Addr basics.Address
+
+	AccountData
+}
+
+// NewAccountDeltas stores ordered accounts and allows fast lookup by address
+type NewAccountDeltas struct {
 	// Actual data. If an account is deleted, `accts` contains a balance record
 	// with empty `AccountData`.
-	accts []basics.BalanceRecord
+	accts []newBalanceRecord
 	// cache for addr to deltas index resolution
 	acctsCache map[basics.Address]int
+
+	appParams      map[AccountApp]*basics.AppParams
+	appLocalStates map[AccountApp]*basics.AppLocalState
+	assetParams    map[AccountAsset]*basics.AssetParams
+	assets         map[AccountAsset]*basics.AssetHolding
```

Comment:
you use pointers here but `cachedAppParams.exists` struct/bool wrapped objects for roundCowBase, is there a difference?

Reply:
yes, nil is an indication a resource was deleted, the cache you refer to is for base data from the underlying acct updates and DB. I store boolean field is just not to query acct updates/db again for non-existing data.

Reply:
aha, right, got it